kratos/processes/apply_periodic_boundary_condition_process.cpp

```diff
@@ -0,0 +1,357 @@
+                                  Parameters rParameters) : Process(Flags()), mrMasterModelPart(rMasterModelPart),
+                                   mrSlaveModelPart(rSlaveModelPart), mParameters(rParameters)
+    {
+        Parameters default_parameters(R"(
+                                            {
+                                            "variable_names":[],
+                                            "center":[0,0,0],
+                                            "axis_of_rotation":[0.0,0.0,0.0],
+                                            "angle":0.0,
+                                            "dir_of_translation":[0.0,0.0,0.0],
+                                            "magnitude":0.0
+                                            }  )");
+
+        // Initializing
+        mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+        mCenterOfRotation.push_back(mParameters["center"][0].GetDouble());
+        mCenterOfRotation.push_back(mParameters["center"][1].GetDouble());
+        mCenterOfRotation.push_back(mParameters["center"][2].GetDouble());
+
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][0].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][1].GetDouble());
+        mAxisOfRotationVector.push_back(mParameters["axis_of_rotation"][2].GetDouble());
+
+        mModulus = mParameters["magnitude"].GetDouble();
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][0].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][1].GetDouble());
+        mDirOfTranslation.push_back(mParameters["dir_of_translation"][2].GetDouble());
+
+        mTheta = mParameters["angle"].GetDouble() * 2 * 3.1416 / 360.0;
+
+        mTransformationMatrix.resize(4,4);
+        mTransformationMatrixVariable.resize(4,4);
+
+        RemoveCommonNodesFromSlaveModelPart();
+
+        if (mTheta == 0 && mModulus != 0)
+            mType = "translation";
+        else if (mTheta != 0.0 && mModulus == 0.0)
+            mType = "rotation";
+        else
+            KRATOS_ERROR_IF(mTheta == 0.0 && mModulus == 0.0)<<"Both angle of rotation and modulus of translation cannot be zero. Please check the input"<<std::endl;
+        else
+            KRATOS_ERROR_IF(mTheta != 0.0 && mModulus != 0.0)<<"Both angle of rotation and modulus of translation cannot be specified at the same time. Please check the input"<<std::endl;
+
+        CalculateTransformationMatrix();
+        mIsInitialized = false;
+    }
+
+
+    ApplyPeriodicConditionProcess::~ApplyPeriodicConditionProcess()
+    {
+    }
+
+    /**
+     * @brief  Function to remove the common nodes of slave and master modelparts from the slave modelpart
+     */
+    void ApplyPeriodicConditionProcess::RemoveCommonNodesFromSlaveModelPart()
+    {
+        const int num_slave_nodes = mrSlaveModelPart.NumberOfNodes();
```

Comment:
**Big question:** how does this work / is intended to work in MPI?
If there should only be one of these processes then I think you should apply some operations only on local nodes

Reply:
this was not addresses

Reply:
I commented on @msandre 's comments .. ! we can discuss there !

Reply:
ok lets discuss this once the MPCs are working in MPI